docs/develop/memorymgt.rst

```diff
@@ -0,0 +1,131 @@
+.. _memorymanagement:
+
+Memory management
+=================
+
+Unlike many programming languages such as C/C++, MicroPython hides memory management 
+details from the developer by supporting automatic memory management.
+Automatic memory management is a technique used by operating systems or applications to automatically manage 
+the allocation and deallocation of memory. This eliminates challenges such as forgetting to
+free the memory allocated to an object. Automatic memory management also avoids the critical issue of using memory
+that is already released. Automatic memory management takes many forms, one of them being
+garbage collection (GC).
+
+The garbage collector usually has two responsibilities;
+
+#. Allocate new objects in available memory.
+#. Free unused memory.
+
+There are many GC algorithms but MicroPython uses the 
+`Mark and Sweep <https://www.geeksforgeeks.org/mark-and-sweep-garbage-collection-algorithm/>`_
+policy for managing memory. This algorithm has a mark phase that traverses the heap marking all
+live objects while the sweep phase goes through the heap reclaiming all unmarked objects.
+
+.. note::
+   The garbage collector automatically runs on the Linux port but may need to be manually 
+   enabled for other ports.
+
+Garbage collection functionality in MicroPython is available through the ``gc`` built-in
+module:
+
+.. code-block:: console
+   
+   >>> x = 5
+   >>> x
+   5
+   >>> import gc
+   >>> gc.enable()
+   >>> gc.mem_alloc()
+   1312
+   >>> gc.mem_free()
+   2071392
+   >>> gc.collect()
+   19
+   >>> gc.disable()
+   >>> 
+
+Even when ``gc.disable()`` is invoked, collection can be triggered with ``gc.collect()``.
+
+The object model
+-----------------
+
+The structure of a MicroPython object is such that is takes up a word-size. That is to say, 
+pointers and addresses take up a machine word of 8 bytes. Pointers can be 8, 16, or 64 bits.
```

Comment:
Only 8-bytes on 64-bit (which is pretty much only the Unix port).

Perhaps clairfy that this section is specifically talking about OBJ_REPR_D, which is used on Unix 64-bit. See mpconfig.h for the other representations.